aten/src/ATen/native/Distance.cpp

```diff
@@ -52,6 +53,21 @@ Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
   return result;
 }
 
+Tensor _cdist_backward(const Tensor& grad, const Tensor& x1, const Tensor& x2, const double p, const Tensor& dist) {
+  AT_CHECK(x1.is_contiguous(), "_cdist_backward requires X1 to be contiguous");
```

Comment:
A backward function should do `.contiguous` rather than check if an input is contiguous because the input is often fed by another backward function